cmake/inputs/DownloadPrebuiltTileDB.cmake

```diff
@@ -0,0 +1,137 @@
+                        )
+        else()
+                message(WARNING "Downloading release list without SHA checksum!")
+                file(DOWNLOAD
+                        ${UPSTREAM_URL}/${VERSION}/rellist.csv
+                        rellist.csv
+                        SHOW_PROGRESS
+                        )
+        endif()
+
+        file(STRINGS
+                ${CMAKE_CURRENT_BINARY_DIR}/rellist.csv
+                RELLIST
+                )
+
+        # Remove csv table headers
+        list(POP_FRONT RELLIST)
+
+        foreach(LINE ${RELLIST})
+                string(REPLACE "," ";" LINE ${LINE})
+                list(LENGTH LINE LENGTH)
+
+                list(GET LINE 0 OS)
+                list(GET LINE 1 ARCH)
+                list(GET LINE 2 URL)
+                list(GET LINE 3 SHA)
+
+                if(${LENGTH} GREATER 4)
+                        list(GET LINE 4 NOAVX2)
+                endif()
+
+                set(RELEASE_VAR TILEDB_${OS}_${ARCH})
+                set(URL_${RELEASE_VAR} ${URL} PARENT_SCOPE)
+                set(HASH_${RELEASE_VAR} ${SHA} PARENT_SCOPE)
+        endforeach()
+endfunction()
+
+function(detect_artifact_name OUT_VAR)
+        if (WIN32) # Windows
+                SET(${OUT_VAR} TILEDB_WINDOWS_X86_64 PARENT_SCOPE)
+        elseif(APPLE) # OSX
+                if (DEFINED CMAKE_OSX_ARCHITECTURES)
+                        set(ACTUAL_TARGET ${CMAKE_OSX_ARCHITECTURES})
+                else()
+                        set(ACTUAL_TARGET ${CMAKE_SYSTEM_PROCESSOR})
+                endif()
+
+
+                if (ACTUAL_TARGET MATCHES "(x86_64)|(AMD64|amd64)|(^i.86$)")
+                        SET(${OUT_VAR} TILEDB_MACOS_X86_64 PARENT_SCOPE)
+                elseif (ACTUAL_TARGET STREQUAL arm64 OR ACTUAL_TARGET MATCHES "^aarch64" OR CMAKE_SYSTEM_PROCESSOR MATCHES "^arm")
+                        SET(${OUT_VAR} TILEDB_MACOS_ARM64 PARENT_SCOPE)
+                endif()
+        else() # Linux
+                SET(${OUT_VAR} TILEDB_LINUX_X86_64 PARENT_SCOPE)
+        endif()
+endfunction()
+
+function(get_tiledb_source_url_and_hash OUT_URL OUT_HASH)
+        # TODO
+endfunction()
+
+function(fetch_prebuilt_tiledb)
+        # Arguments
+        set(options RELLIST_HASH)
+        set(oneValueArgs VERSION ARTIFACT_NAME)
+        set(multiValueArgs)
+        cmake_parse_arguments(
+                FETCH_PREBUILT_TILEDB
+                "${options}"
+                "${oneValueArgs}"
+                "${multiValueArgs}"
+                ${ARGN}
+                )
+
+        if(NOT ${FETCH_PREBUILT_TILEDB_VERSION})
+                set(FETCH_PREBUILT_TILEDB_VERSION "{{LATEST_TILEDB_VERSION}}")
+        endif()
+        fetch_tiledb_release_list(${FETCH_PREBUILT_TILEDB_VERSION} RELLIST_HASH)
+
+        if(NOT ${FETCH_PREBUILT_TILEDB_ARTIFACT_NAME})
+                detect_artifact_name(FETCH_PREBUILT_TILEDB_ARTIFACT_NAME)
+        endif()
+
+        ExternalProject_Add(ep_tiledb
```

Comment:
Use `FetchContent` or plain downloading?

Reply:
I don't think that this PR should rework this download logic from the original approach.

Reply:
I am concerned that `ExternalProject` might force downstream projects to use a superbuild architecture.

Reply:
> ExternalProject

Is this something we should address now in your opinion?

Reply:
Hmm, the issue is that they still use it right? Don't we want to switch to vcpkg style dependencies? (ports)

Reply:
I think the plan is that all first-party downstream projects eventually get TileDB from vcpkg.

We might want to change `ExternalPorject` later, but it will be a breaking change which might or might not be OK to do.